app/views/user_mailer/new_device_sign_in_attempt.html.erb

```diff
@@ -0,0 +1,37 @@
+<% t(:info_html, scope: @i18n_scope).each do |info_p| %>
+  <p class="margin-bottom-2">
+    <%= I18n.interpolate(
+          info_p,
+          reset_password_link_html: link_to(
+            t(:reset_password, scope: @i18n_scope),
+            event_disavowal_url(disavowal_token: @disavowal_token),
+          ),
+        ).html_safe %>
+  </p>
+<% end %>
+
+<% @events.map { |event| [IpGeocoder.new(event.device.last_ip), event] }.
+     group_by { |geocoder,| geocoder.location }.
+     each do |location, geocoder_events| %>
```

Comment:
what if we moved this grouping logic to the mailer class?
```suggestion
<% @events_by_location.each do |location, geocoder_events| %>
```

Reply:
Yeah that'd probably be better 👍

Reply:
I pushed this in 9ca5d56 but based on some [discussion](https://gsa-tts.slack.com/archives/C05R6BLVAQG/p1711546350252369?thread_ts=1711476471.048729&cid=C05R6BLVAQG) I think we're going to revert to using Eastern time as a default, which removes the need for this. Replaced in 544de0d.